software/workspace/workspace.go

```diff
@@ -193,39 +199,42 @@ func getWorkspaceSelection(pageSize, pageNumber int, client houston.ClientInterf
 
 	tabPrintErr := tab.PrintWithPageNumber(pageNumber*pageSize, out)
 	if tabPrintErr != nil {
-		return "", fmt.Errorf("unable to print with page number: %w", tabPrintErr)
+		return workspaceSelection{id: "", quit: false, err: fmt.Errorf("unable to print with page number: %w", tabPrintErr)}
 	}
 	totalRecords := len(ws)
 
 	if pageSize > 0 {
 		selectedOption := workspacesPromptPaginatedOption(pageSize, pageNumber, totalRecords)
 		if selectedOption.quit {
 			if selectedOption.userSelection == 0 {
-				return "", nil
+				return workspaceSelection{id: "", quit: true, err: nil}
 			}
-			return ws[selectedOption.userSelection-1].ID, nil
+			return workspaceSelection{id: ws[selectedOption.userSelection-1].ID, quit: false, err: nil}
 		}
 		return getWorkspaceSelection(selectedOption.pageSize, selectedOption.pageNumber, client, out)
 	}
 
 	in := input.Text("\n> ")
 	i, err := strconv.ParseInt(in, 10, 64) //nolint:gomnd
 	if err != nil {
-		return "", fmt.Errorf("cannot parse %s to int: %w", in, err)
+		return workspaceSelection{id: "", quit: false, err: fmt.Errorf("cannot parse %s to int: %w", in, err)}
 	}
-
-	return ws[i-1].ID, nil
+	return workspaceSelection{id: ws[i-1].ID, quit: false, err: nil}
 }
 
 // Switch switches workspaces
 func Switch(id string, pageSize int, client houston.ClientInterface, out io.Writer) error {
 	if id == "" {
-		_id, err := getWorkspaceSelection(pageSize, 0, client, out)
-		if err != nil {
-			return err
+		workspaceSelection := getWorkspaceSelection(pageSize, 0, client, out)
+
+		if workspaceSelection.quit {
+			return nil
```

Comment:
can we add unit tests for this case

Reply:
Added test cases for workspace switch cmd